packages/protocol-sdk/src/1155-create-helper.ts

```diff
@@ -0,0 +1,319 @@
+import {
+  zoraCreator1155FactoryImplABI,
+  zoraCreator1155FactoryImplAddress,
+  zoraCreator1155ImplABI,
+  zoraCreatorFixedPriceSaleStrategyABI,
+} from "@zoralabs/protocol-deployments";
+import type {
+  Address,
+  Hex,
+  PublicClient,
+  TransactionReceipt,
+  WalletClient,
+} from "viem";
+import { decodeEventLog, encodeFunctionData, zeroAddress } from "viem";
+import { OPEN_EDITION_MINT_SIZE } from "./constants";
+
+const saleEndForever = 18446744073709551615n;
+const royaltyBPSDefault = 1000;
+
+type SalesConfigParamsType = {
+  // defaults to 0
+  pricePerToken?: bigint;
+  // defaults to 0, in seconds
+  saleStart?: bigint;
+  // defaults to forever, in seconds
+  saleEnd?: bigint;
+  maxTokensPerAddress?: number;
+  fundsRecipient?: Address;
+};
+
+export const DEFAULT_SALE_SETTINGS = {
+  fundsRecipient: zeroAddress,
+  pricePerToken: 0n,
+  saleStart: 0n,
+  saleEnd: saleEndForever,
+  maxTokensPerAddress: 0n,
+};
+const PERMISSION_BIT_MINTER = 2n ** 2n;
+
+type ContractType =
+  | {
+      name: string;
+      uri: string;
+      defaultAdmin?: Address;
+    }
+  | Address;
+
+type RoyaltySettingsType = {
+  royaltyBPS: number;
+  royaltyRecipient: Address;
+};
+
+export function create1155TokenSetupArgs({
+  nextTokenId,
+  mintToCreatorCount,
+  tokenMetadataURI,
+  fixedPriceMinterAddress,
+  createReferral,
+  maxSupply,
+  account,
+  salesConfig,
+  royaltySettings,
+}: {
+  maxSupply?: bigint | number;
```

Comment:
how about just requiring a bigint here and making things simpler? (prevent it from being a number)